File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnPlaylistResult.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* <b>Experimental</b>: GnPlaylistResult
*/
public class GnPlaylistResult : GnObject {
  private HandleRef swigCPtr;

  internal GnPlaylistResult(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnPlaylistResult_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnPlaylistResult obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnPlaylistResult() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnPlaylistResult(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnPlaylistResultIdentEnumerable Identifiers {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnPlaylistResult_Identifiers_get(swigCPtr);
      GnPlaylistResultIdentEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnPlaylistResultIdentEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnPlaylistResult.java

package com.gracenote.gnsdk;

/** 
* <b>Experimental</b>: {@link GnPlaylistResult} 
*/ 
 
public class GnPlaylistResult extends GnObject {
  private long swigCPtr;

  protected GnPlaylistResult(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnPlaylistResult_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnPlaylistResult obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnPlaylistResult(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnPlaylistResultIdentIterable identifiers() {
    return new GnPlaylistResultIdentIterable(gnsdk_javaJNI.GnPlaylistResult_identifiers(swigCPtr, this), true);
  }

}
